Sept 2023 updates

fivetran-reneeli opened the pull request:
## PR Overview
**This PR will address the following Issue/Feature:**

https://github.com/fivetran/dbt_klaviyo/issues/28

**This PR will result in the following new package version:**
<!--- Please add details around your decision for breaking vs non-breaking version upgrade. If this is a breaking change, were backwards-compatible options explored? -->

v0.7.0

Removing deprecated and adding new fields

**Please detail what change(s) this PR introduces and any additional information that should be known during the review of this PR:**

Following these [release notes](https://fivetran.com/docs/applications/klaviyo/changelog) for changes made to the connector.

## PR Checklist
### Basic Validation
Please acknowledge that you have successfully performed the following commands locally:
- [x] dbt compile
- [x] dbt run –full-refresh
- [x] dbt run
- [x] dbt test
- [ ] dbt run –vars (if applicable)

Before marking this PR as "ready for review" the following have been applied:
- [x] The appropriate issue has been linked and tagged
- [x] You are assigned to the corresponding issue and this PR
- [x] BuildKite integration tests are passing

### Detailed Validation
Please acknowledge that the following validation checks have been performed prior to marking this PR as "ready for review":
- [x] You have validated these changes and assure this PR will address the respective Issue/Feature.
- [ ] You are reasonably confident these changes will not impact any other components of this package or any dependent packages.
- [ ] You have provided details below around the validation steps performed to gain confidence in these changes.
<!--- Provide the steps you took to validate your changes below. -->

- dbt docs show updated ymls
- downstream models succeed 

### Standard Updates
Please acknowledge that your PR contains the following standard updates:
- Package versioning has been appropriately indexed in the following locations:
    - [x] indexed within dbt_project.yml
    - [x] indexed within integration_tests/dbt_project.yml
- [x] CHANGELOG has individual entries for each respective change in this PR
    <!--- If there is a parallel upstream change, remember to reference the corresponding CHANGELOG as an individual entry.  -->
- [ ] README updates have been applied (if applicable)
    <!--- Remember to check the following README locations for common updates. →
        <!--- Suggested install range (needed for breaking changes) →
        <!--- Dependency matrix is appropriately updated (if applicable) →
        <!--- New variable documentation (if applicable) -->
- [ ] DECISIONLOG updates have been updated (if applicable)
- [ ] Appropriate yml documentation has been added (if applicable)

### dbt Docs
Please acknowledge that after the above were all completed the below were applied to your branch:
- [ ] docs were regenerated (unless this PR does not include any code or yml updates)

### If you had to summarize this PR in an emoji, which would it be?
<!--- For a complete list of markdown compatible emojis check our this git repo (https://gist.github.com/rxaviers/7360908)  --> 
:dancer:

fivetran-reneeli commented:
Will wait on regen docs til after PR approved

fivetran-jamie commented on models/stg_klaviyo__campaign.sql:
```diff
@@ -31,14 +31,15 @@ final as (
         status_id,
         subject,
         updated as updated_at,
+        archived,
```
since this is a boolean do we wanna rename it to `is_archived`?

fivetran-reneeli replied:
Ah thanks Jamie-- just realized I had some changes that weren't pushed, this included

fivetran-jamie commented on models/stg_klaviyo__flow.sql:
```diff
@@ -29,14 +29,9 @@ final as (
-        as flow_trigger,
         updated as updated_at,
-        customer_filter as person_filter,
+        archived,
```
same comment as above -- should we rename to `is_archived`?

fivetran-reneeli replied:
udpated

fivetran-jamie commented on models/stg_klaviyo__person.sql:
```diff
@@ -42,6 +42,7 @@ final as (
         timezone,
         title,
         updated as updated_at,
+        last_event_date as last_event_date_at,
```
is the `date_at` part redundant? wondering if we should leave it as `last_event_date` or rename it to just `last_event_at`

fivetran-reneeli replied:
good idea, renamed to last_event_at

fivetran-jamie commented on CHANGELOG.md:
```diff
@@ -1,3 +1,29 @@
+
+| **Table**                          | **Column**                                                                                                                                                                                                                             |
+|--------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| CAMPAIGN | archived
+| CAMPAIGN | scheduled
+| FLOW | archived
+| FLOW | trigger_type
+| PERSON | last_event_date
```
i feel like we should have a third column that shows what we rename these fields to (if we rename them) in the respective staging models

fivetran-jamie replied:
also could be good to add DAG links to the affected source tables so people can quickly see the descriptions for these new fields

fivetran-jamie replied:
same for the removed fields

fivetran-reneeli replied:
awesome, updated!

fivetran-jamie commented on CHANGELOG.md:
```diff
@@ -1,3 +1,29 @@
+
+## 🚨 Breaking Changes 🚨:
+
+- We have removed these columns from the following tables:
```
```suggestion
- We have removed these deprecated columns from the following tables:
```

fivetran-jamie reviewed (changes requested):
looks pretty good! nothing big stands out, but left some style- and doc-related comments

fivetran-jamie reviewed (commented):
looks great! will approve after some minor things in the changelog are addressed and docs are regenerated

fivetran-jamie commented on models/stg_klaviyo__person.sql:
```diff
@@ -42,6 +42,7 @@ final as (
         timezone,
         title,
         updated as updated_at,
+        last_event_date as last_event_at,
```
ah actually this is causing ambiguous column errors in the final persons model due to https://github.com/fivetran/dbt_klaviyo/blob/main/models/intermediate/int_klaviyo__person_metrics.sql#L17

may want to leave it as `last_event_date` or consider removing it?

fivetran-reneeli replied:
Oh good noticing, I didn't realize the downstream error. I feel like it's still worthwhile adding it, so will just keep it as `last_event_date`. Thanks Jamie!

fivetran-reneeli commented on CHANGELOG.md:
```diff
@@ -1,3 +1,35 @@
+| FLOW | customer_filter
+| FLOW | trigger
+
+- We have added these columns from the following tables:
```
ah good reminder

fivetran-reneeli commented:
Thanks @fivetran-jamie! Updated everything and regen docs. I also realized that email in the persons staging may not necessarily exist, based off testing off the internal schema, so I removed the not null test.

fivetran-jamie commented on CHANGELOG.md:
```diff
@@ -1,3 +1,36 @@
+| CAMPAIGN | scheduled | scheduled_at
+| FLOW | archived | is_archived
+| FLOW | trigger_type | trigger_type
+| PERSON | last_event_date | last_event_at
```
```suggestion
| PERSON | last_event_date | last_event_date
```

fivetran-jamie reviewed (approved):
lgtm! just one thing in the changelog but otherwise this is approved
